### PR TITLE
Adding Logging Facility to Lamp

### PR DESCRIPTION
**Pull request by prakhar-mudaiya**
Adding Logging Facility to Lamp :- 
1. To add logging put logPath in config
2. Logging to file will be done for every verbose and print message
3. Logging to file has 3 level INFO, WARN and ERROR
4. If no logPath is specified logging to file will not happen. Their will be a message " Logging to file is disabled, To enable Logging to file Please specify logPath in configuration"
5. Tested changes for all alerts, escalation and incident commands

### CODE REVIEW

File: command/command.go [hunk 1]
```diff
@@ -28,16 +29,51 @@ import (
 
 var verbose = false
 
+type LogLevel string
+
+const (
+	INFO LogLevel = "INFO"
+	WARN LogLevel = "WARN"
```
Comment (cemkucuk) [inline command/command.go]: If "WARN" is not using now, we can remove the line

File: command/command.go [hunk 2]
```diff
@@ -28,16 +29,51 @@ import (
...
+
 func printVerboseMessage(message string) {
 	if verbose {
 		fmt.Println(message)
 	}
+	logToFile(INFO, message)
 	logger := client.Config{}.Logger
 	if logger != nil {
 		logger.Debug(message)
 	}
```
Comment (cemkucuk) [inline command/command.go]: can call new logToFile method for this debug message from SDK

File: command/command.go [hunk 3]
```diff
@@ -28,16 +29,51 @@ import (
...
 }
 
+/*
+This method Print the message to console and Logfile based on corresponding Log Level
+ */
+func printMessage(logLevel LogLevel, message string) {
+	log.Println(string(logLevel) + " : " + message)
+	logToFile(logLevel, message)
+}
```
Comment (cemkucuk) [inline command/command.go]: Why we can't use printMessage instead of printVerboseMessage?

File: command/command.go [hunk 4]
```diff
@@ -28,16 +29,51 @@ import (
...
+	logToFile(INFO, message)
 	logger := client.Config{}.Logger
 	if logger != nil {
 		logger.Debug(message)
 	}
 }
 
+/*
+This method Print the message to console and Logfile based on corresponding Log Level
+ */
```
Comment (cemkucuk) [inline command/command.go]: we're not using function describe comments any parts of our codebase fyi

File: command/command.go [hunk 5]
```diff
@@ -51,6 +87,15 @@ func grabAPIKey(c *gcli.Context) string {
 	return apiKey
 }
 
+func grabLogPath() string {
```
Comment (cemkucuk) [inline command/command.go:85]: This function called for every log action to fetch path from configuration, imo we should call it at one in the beginning

**Review by cemkucuk — CHANGES_REQUESTED** [review]
@prakhar-mudaiya can you look at the comments?

File: command/command.go (hunk 1 above)
Comment (prakhar-mudaiya) [reply command/command.go]: Removed WARN

File: command/command.go (hunk 2 above)
Comment (prakhar-mudaiya) [reply command/command.go]: We already had a discussion. This logger is coming from go config. Hence did not change it

File: command/command.go (hunk 3 above)
Comment (prakhar-mudaiya) [reply command/command.go]: Made the changes and removed printVerboseMessage. Create a new Log Level Debug. which will be used verbose

File: command/command.go (hunk 4 above)
Comment (prakhar-mudaiya) [reply command/command.go]: Removed from all places

File: command/command.go (hunk 5 above)
Comment (prakhar-mudaiya) [reply command/command.go:85]: Done

File: command/command.go [hunk 6]
```diff
@@ -28,13 +29,43 @@ import (
...
+	INFO LogLevel = "INFO"
+	ERROR LogLevel = "ERROR"
+	DEBUG LogLevel = "DEBUG"
+)
+
+func printMessage(logLevel LogLevel, message string) {
+	if logLevel == DEBUG {
+		if verbose {
+			log.Println(string(logLevel) + " : " + message)
+			logToFile(DEBUG, message)
```
Comment (cemkucuk) [inline command/command.go:46]: I think we should put logToFile method call to outside of verbose flag

**Review by cemkucuk — CHANGES_REQUESTED** [review]

**Review by cemkucuk — APPROVED** [review]

**Pull request merged**